api/src/main/java/org/openmrs/module/reporting/cohort/definition/evaluator/ConditionCohortDefinitionEvaluator.java

```diff
@@ -0,0 +1,70 @@
+package org.openmrs.module.reporting.cohort.definition.evaluator;
+
+import java.util.List;
+
+import org.openmrs.Cohort;
+import org.openmrs.annotation.Handler;
+import org.openmrs.module.reporting.cohort.EvaluatedCohort;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.ConditionCohortDefinition;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.evaluation.querybuilder.SqlQueryBuilder;
+import org.openmrs.module.reporting.evaluation.service.EvaluationService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+@Handler(supports = { ConditionCohortDefinition.class })
+public class ConditionCohortDefinitionEvaluator implements CohortDefinitionEvaluator {
+	
+	@Autowired
+	EvaluationService evaluationService;
+	
+	@Override
+	public EvaluatedCohort evaluate(CohortDefinition cohortDefinition, EvaluationContext context) {
+		
+		ConditionCohortDefinition cd = (ConditionCohortDefinition) cohortDefinition;
+		StringBuilder sql = new StringBuilder();
+		SqlQueryBuilder query = new SqlQueryBuilder();
+		
+		String sqlDefault = "SELECT patient_id FROM conditions c";
+		String sqlDate = "";
+		String sqlConditonValue = "";
+		sql.append(sqlDefault);
+		if (cd.getOnOrAfter() != null && !(cd.getOnOrBefore() != null)) {
+			sqlDate = " WHERE c.date_created >= :onOrAfter";
+		}
+		if (cd.getOnOrBefore() != null && !(cd.getOnOrAfter() != null)) {
+			sqlDate = " WHERE c.date_created <= :onOrBefore ";
+		}
+		if (cd.getOnOrBefore() != null && cd.getOnOrAfter() != null) {
+			sqlDate = " WHERE c.date_created >= :onOrAfter AND c.date_created <= :onOrBefore ";
```

Comment:
All of this would be a lot simpler (and is preferred), if you were to use the HqlQueryBuilder instead.  There are a few things I notice here:  1. it would be better to check and apply onOrAfter and onOrBefore once each, instead of multiple boolean permutations of each.  2. it is likely that your comparison of "<: onOrBefore" will inadvertently exclude some patients on that day, since most date parameters are not timestamps, and have timestamps set to midnight at the beginning of the day.

Reply:
@mseaton , i innitially used the HQLQuery builder ,and here are the few issues that came up
- I had to pass the Conditon Class as a param in the HQLQuery builder.
That required either to use the Condition Class in EmrApi or in Core 2.2.

Depending on the Class in the core meant upgrading the core dependence to 2.2 , but that resulted into alot many other errors which i ddint want to go through atleast for now.
Thats when i tried depending on EmrApi-conditon-List.
However , since that Conditon class  in EmrApi ,is not in the Core, it couldnt load the Hibernate mappings by default,and was resulting in errors like "Condition Class is not mapped" , so thats when i decided to take the easier option , to instead use the SQLQuery builder as i needed not to pass the Condition Class anywhere